reservation/reservation.go

```diff
@@ -106,6 +106,16 @@ func (manager *Manager) Lock(resource string) (*Reservation, error) {
 	return res, nil
 }
 
+// WaitUntilLock creates a Reservation for `resource`, or waits until it can do so.
+func (manager *Manager) WaitUntilLock(resource string) (*Reservation, error) {
+	res, err := manager.Lock(resource)
+	for err != nil && err.Error() == fmt.Sprintf("Reservation already exists for resource %s", resource) {
```

Comment:
It might be worth putting that into a function for more clarity, something like:

```go
func reservationAlreadyExists(err error, resource string) bool {
  return fmt.Sprintf("%s", err) == fmt.Sprintf("Reservation already exists for resource %s", resource)
}
```

Then your loop can look like:

```go
for res, err = manager.Lock(resource); reservationAlreadyExists(err, resource); res, err = manager.Lock(resource) {
  time.Sleep(time.Second)
}
```